src/odoc/occurrences.ml

```diff
@@ -0,0 +1,205 @@
+open Or_error
+
+let handle_file file ~f =
+  Odoc_file.load file
+  |> Result.map @@ fun unit' ->
+     match unit' with
+     | { Odoc_file.content = Unit_content unit; _ } -> Some (f unit)
+     | _ -> None
+
+let fold_dirs ~dirs ~f ~init =
+  dirs
+  |> List.fold_left
+       (fun acc dir ->
+         acc >>= fun acc ->
+         Fs.Directory.fold_files_rec_result ~ext:"odocl"
+           (fun acc file ->
+             file |> handle_file ~f:(f acc) >>= function
+             | None -> Ok acc
+             | Some acc -> Ok acc)
+           acc dir)
+       (Ok init)
+
+module H = Hashtbl.Make (Odoc_model.Paths.Identifier)
+
+module Occtbl : sig
+  type item = { direct : int; indirect : int; sub : item H.t }
+  type t = item H.t
+  type key = Odoc_model.Paths.Identifier.t
+  val v : unit -> t
+
+  val add : t -> key -> unit
+
+  val iter : (key -> item -> unit) -> t -> unit
+
+  val get : t -> key -> item option
+end = struct
+  type item = { direct : int; indirect : int; sub : item H.t }
+  type t = item H.t
+  type key = Odoc_model.Paths.Identifier.t
+
+  let v_item () = { direct = 0; indirect = 0; sub = H.create 0 }
+
+  let v () = H.create 0
+
+  let add tbl id =
+    let rec add ?(kind = `Indirect) id =
+      let incr htbl id =
+        let { direct; indirect; sub } =
+          match H.find_opt htbl id with Some n -> n | None -> v_item ()
```

Comment:
Compat with 4.02:
```suggestion
          try H.find htbl id with Not_found -> v_item ()
```

Reply:
I'll wait for #1060 for those kind of issues.